### ISSUE TITLE
plot,section-method() help page missing information for `which` argument

### ISSUE THREAD

**Issue by avictor88** [dankelley/oce#2136]
It isn’t clear that plots can be made with data that isn’t included in the list of plot options under “Details” (eg. if the data is available, plots of fluorescence and pH can be made even though they aren’t listed as options for `which`). The help page could be improved by explaining that any data in a section object can be plotted using `plot,section-method()` by writing out the data name in the `which` argument.

**Comment by dankelley** [comment]
This is a good idea.  It would only take me a minute or two to do the change, but it might be good for you to learn how to do a "pull request", if that's something you have heard about and want to learn.

To do this, you'd want to "fork" the oce package, and work in the branch named "develop".

Then, find the relevant file in the `R` directory.  In that file, find the line containing
```R
#' @param which
```
that corresponds to the "plot" function.  (Caution: that line might occur in several places in the file, so you need to look at adjacent text to be sure you're at the right spot.)

Then add a sentence or phrase, as appropriate.

Then build the package locally.  Be sure that your build setup is configured to create documentation, because then the relevant file in the `man` directory would also be updated.

Building will take 1 to 3 minutes.  If you don't have a C compiler on your machine, just give up and I'll do this.

In your build setup, be sure that the configuration indicates to RStudio to update the documentation.  This is needed to build a file in the `man` directory.

If the build works, do a "check" operation in RStudio.  That will take maybe 5 minutes.  If anything fails, let me know.

Then use the help system to read through your new text,  to see if (a) the changes "took" and if (b) you like the new wording.

I can just go ahead and do this myself, if you don't feel inclined to make a pull request.  Certainly, that's not a problem because the work of my doing that is maybe 1/10-th of the work it took me to write this message.  I just thought that learning how to do pull requests is a great skill for you.

PS. I can't remember whether you know how to do PRs already.  If you do, I apologize for this long comment :-)

**Comment by richardsc** [comment]
I agree this would be good for @avictor88 to learn, though we haven't had a lot of time to include git instruction in our summer work. I'm meeting with her in about 15 minutes and we'll discuss 😄

**Comment by avictor88** [comment]
Thanks for the explanation Dan! I've been working on this to learn how to do PRs, and all of the steps have been going well up to the point of checking the built package. I've attached a screenshot of the error that I'm getting, and I'll go over my steps that led up to this to see if there's somewhere that I went wrong.

After forking "oce" and editing the "section.R" file in the "develop" branch, I cloned the edited "oce" repository from GitHub to my computer and created a project in RStudio from this directory. I then built the package using "Install and Restart" after making sure that it was configured to create files. When the build was complete, I tried to check it, but I got the attached error. I also tried to load in the new "oce" package to see if my changes took, but they didn't show up, and when I looked into the "man" directory there were no new or updated files.

The error that I got doesn't seem to be related to my edits, but I'm still curious as to why I'm not seeing any of my changes on the help page. If I made any mistakes let me know, and thanks for taking the time to help me learn how to do this!

![Screen Shot 2023-08-18 at 11 21 56 PM](https://github.com/dankelley/oce/assets/132913765/3f4530b5-3503-4ed6-a7a9-5b8d7823cf5b)

**Comment by dankelley** [comment]
Hi Anna.  First, you posted this at midnight -- you work far too hard!!

Second, The error (red text) tells you that the package named `roxygen2` is not found.  That means that it is not installed.  The way R works is that there is a base set of software ,and then a bunch (tens of thousands) of "packages" that do extra things.  For example, `oce` is a package.  The way to install packages is to use the `install.packages()` function.

Thus, you ought to do
```R
install.packages("roxygen2")
```
and then retry.  

You may find that you get red text (errors) for other packages.  You'll need to install them all.  Note that once they are installed, you won't need to re-install them later (unless R undergoes a major new version, which happens every few years).  There is also a function that lists all the packages in your system.  Some people run that, and save the output into an R file, and then wrap that output in a `install.packages()` call, so that they can get their packages installed after a new R version becomes available.  I don't do that myself, anymore, because I like being reminded of which packages I need to run my R scripts.

I do hope you don't read this until Monday!

**Comment by avictor88** [comment]
Oops, silly mistake - that's what I get for my weird work hours :) but at least it has an easy solution! I was able to install all of the necessary packages to build and check oce - everything ran smoothly and the help page had the edits I made. Now that I’ve checked things on my end, I created a PR to merge my branch to the develop branch of oce. If there are any issues with this let me know, but hopefully this closes the issue.

I really appreciate you taking the time to help me learn how to do PRs, and for the run down on packages in R, it’s all great to know!
